packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,124 @@
+import type { Rule } from 'eslint';
+
+import { shorthandFor } from './utils';
+
+const findProgramRoot = (node: Rule.Node) => {
+  const parent = node.parent;
+  if (!parent) {
+    return null;
+  }
+  if (parent.type === 'Program') {
+    return parent;
+  }
+  return findProgramRoot(parent);
+};
+
+export const shorthandFirst: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
+        'When using both shorthand and longhand properties, compiled scrambles the properties and it is unintuative to understand which properties override which others. Ordering the shorthand first ensures that longhand overrides shorthand.',
+      url: 'https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/shorthand-property-sorting',
+    },
+    messages: {
+      'shorthand-first':
+        'When using both shorthand and longhand properties, the shorthand property should be first.',
+    },
+    type: 'problem',
+    fixable: 'code',
+  },
+  create(context) {
+    const selectorString: string =
```

Comment:
we probably don't want to enforce shorthand properties coming before longhand properties for non-Compiled libraries (unless @kylorhall-atlassian has a different opinion)

you can use the `isCss` and `isCssMap` functions in `packages/eslint-plugin/src/utils/is-compiled-import.ts`, and the `isStyled` function from `packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/utils.ts` and `packages/eslint-plugin/src/utils/styled-import.ts`

you'll need to

* create an `isXcss` function
* add `@atlaskit/css` to `COMPILED_IMPORT` in `packages/eslint-plugin/src/utils/constants.ts` (this will involve changing all usages of `COMPILED_IMPORT` so they can handle an array instead of a string)
* maybe move the contents of `packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/utils.ts` to `packages/eslint-plugin/src/utils/is-compiled-import.ts` and potentially refactor to make more consistent


(in an ideal world, we'd probably re-use some of the code that's in the UI Styling Standard ESLint plugin https://atlaskit.atlassian.com/packages/design-system/eslint-utils but oh well)

Reply:
I'm dealing with this from line 47 - 73 with my own checks if I understand what you're saying? I'l check out those functions if that's an easier way to do it though

Reply:
yeah you shouldn't need to implement the logic from scratch, you can extend the existing util functions to do what you need

Reply:
i wouldn't consider this to be a dependency 🤔 

* `@atlaskit/css` is explicitly designed to just be a wrapper around `@compiled/react` so we can treat them as equivalent in this case

* neither we nor DST have shied away from adding [ADS-specific logic](https://atlassian.design/) to Compiled (e.g. the two are [already treated as equivalent in `@compiled/babel-plugin`](https://github.com/atlassian-labs/compiled/blob/b4864f9936d3c0cfbba874cf34ae8bd3b736b816/packages/babel-plugin/src/babel-plugin.ts#L34), all the XCSS stuff)